### ISSUE TITLE
is it intended(and also okay) that dataset_properties of ModelCheckpoint is Dict?

### ISSUE THREAD

**Issue by MilkClouds** [nicolas-chaulet/torch-points3d#655]
Hello, I'am new to torch-point3d and don't know not much about it, but confronted some errors while executing `forward/forward.py` and found the reason.

I ran 
```
poetry run python train.py task=segmentation models=segmentation/pointnet2 model_name=pointnet2_charlesssg data=segmentation/shapenet-fixed
python forward_scripts/forward.py
```

error messsage below occured.
```
File "(mydir)/forward_scripts/../torch_points3d/utils/model_building_utils/model_definition_resolver.py", line 10, in resolve_model
    "FEAT": max(dataset.feature_dimension, 0),
AttributeError: 'dict' object has no attribute 'feature_dimension'
```

Even I edit `data.feature_dimension` in `model_definition_resolver.py` to `data['feature_dimension']`, then `self._num_classes = dataset.num_classes` of `models/segmentation/pointnet2.py/PointNet2_D/init` get stuck. Also after I edit `data.feature_dimension` to `data['feature_dimension']`,  `train.py` get error says that I should edit `data['feature_dimension']` (which I edited) to `data.feature_dimension`, since `feature_dimension` is attiribute of `datasets/base_dataset.py/BaseDataset`.

In conclusion, `datasets/base_dataset.py/BaseDataset` uses feature_dimension as attribute, and 'torch_points3d/metrics/model_checkpoint.py/ModelCheckpoint/dataset_properties' uses feature_dimension as python native dictionary's key. Former needs `abc.feature_dimension` and latter needs 'abc['feature_dimension']`, and those are not compatible, as long as I know and as long as specific source code isn't added.

I think there's some mistake while designing overall source codes. I'm afraid that it's only my misunderstood since I don't know much about this library. I'm curious about other people's opinion.

**Comment by MilkClouds** [comment]
### Solution I'm trying

So, to resolve this problem I tried this things (since I am currently dealing with this problem, this sections will be updated continuously)

Added below source code to `forward_scripts/foward.py`
```
class AttributeDict(dict):
    def __getattr__(self, attr):
        return self[attr]
    def __setattr__(self, attr, value):
        self[attr] = value
```
Also added `checkpoint.dataset_properties = AttributeDict(checkpoint.dataset_properties)` inside main function.

then error below occur. where's weight_classes?...
```
Traceback (most recent call last):
  File "forward_scripts/forward.py", line 91, in main
    model = checkpoint.create_model(checkpoint.dataset_properties, weight_name=cfg.weight_name)
  File "/home/ubuntu/SH/torch-points3d/forward_scripts/../torch_points3d/metrics/model_checkpoint.py", line 174, in create_model
    model = instantiate_model(OmegaConf.create(run_config), dataset)
  File "/home/ubuntu/SH/torch-points3d/forward_scripts/../torch_points3d/models/model_factory.py", line 44, in instantiate_model
    model = model_cls(model_config, "dummy", dataset, modellib)
  File "/home/ubuntu/SH/torch-points3d/forward_scripts/../torch_points3d/models/segmentation/pointnet2.py", line 39, in __init__
    self._weight_classes = dataset.weight_classes
  File "forward_scripts/forward.py", line 56, in __getattr__
    return self[attr]
KeyError: 'weight_classes'
```

**Comment by MilkClouds** [comment]
Copied `predict_original_samples` function from `torch-points3d/torch_points3d/datasets/segmentation/forward/shapenet.py` to
`torch-points3d/torch_points3d/datasets/segmentation/shapenet.py`

error above is solved. but error below occur
```
Traceback (most recent call last):
  File "forward_scripts/forward.py", line 113, in main
    run(model, dataset, device, cfg.output_path)
  File "forward_scripts/forward.py", line 52, in run
    print(dataset.predict_original_samples(data, model.conv_type, model.get_output()))
  File "/home/ubuntu/SH/torch-points3d/forward_scripts/../torch_points3d/datasets/segmentation/shapenet.py", line 381, in predict_original_samples
    sampleid = batch.sampleid[b]
AttributeError: 'SimpleBatch' object has no attribute 'sampleid'
```


where's sampleid? I searched this for all files in this repository, but it seem's like something get wrong..

**Comment by IdeGelis** [comment]
Hi,

I had the same issue when launching the forward script. I found that in `forward.py`, `checkpoint.dataset_properties` is a python  dict while usually configs are 'omegaconf.dictconfig.DictConfig'. In order to get around this issue I added a line in the forward script to convert the `checkpoint.dataset_properties` dict into omegaconf.dictconfig.DictConfig  before the dataset specific configs line 75: 
  >  checkpoint.dataset_properties = OmegaConf.create(checkpoint.dataset_properties)

This is maybe not a proper solution but it is working...

By the way, I didn't get this issue before updating tp3d to the new release 1.3.0. So maybe, this comes from the update of OmegaConf to version 2.0.6.

I hope this help.
Iris
